src/opera/parser/tosca/v_1_3/collector_mixin.py

```diff
@@ -165,6 +166,14 @@ def collect_artifacts(self, service_ast):
         definitions = typ.collect_artifact_definitions(service_ast)
         assignments = self.get("artifacts", {})
 
+        duplicate_interfaces = set(assignments.keys()).intersection(
+            definitions.keys())
+        if duplicate_interfaces:
+            for duplicate in duplicate_interfaces:
+                definitions.pop(duplicate)
+
+        definitions.update(assignments)
```

Comment:
If I understand correctly this is the main change here (most of the others are updating the style)?

Seems OK by me.

Reply:
Yes, here I extended the usage of artifacts so that ones from node templates can be also used. To be sure this works I added some content to the examples so that you can run `opera deploy /examples/artifacts/service.yaml` and the `opera outputs` to see that both file contents were included.

This should be the result:
![image](https://user-images.githubusercontent.com/41727816/95859597-247a9680-0d5f-11eb-9030-1deb2377c386.png)

Reply:
I saw the example, but at first glance, I understood that we want to overload the definition (and therefore re-define the first one). Therefore I thought that the example does not reflect the usage what we want to achieve.  Thanks for your clarification @anzoman !